test/components/date-picker.spec.tsx

```diff
@@ -28,103 +37,102 @@ describe('The DatePicker Component', () => {
-            trigger.change(datePickerInput!, '2017/02/01');
-            simulate.keyDown(datePickerInput, {keyCode: keycode('enter')});
+            driver.datePicker.changeDate('2017/02/01');
 
-            await waitForDom(() => expect(select(currentDate)).to.have.text('Wed Feb 01 2017'));
+            await waitForDom(() => expect(driver.date).to.have.text('Wed Feb 01 2017'));
         });
 
         it('clicks on the input, picks a date from the dropdown, ' +
             'and then expects the dropdown to close and the date to appear in the input', async () => {
-            const {select, waitForDom} = clientRenderer.render(<DatePickerDemo value={JANUARY_FIRST} />);
-            const datePickerInput = select(datePickerInputId);
+            const {driver, waitForDom} = clientRenderer.render(<DatePickerDemo value={JANUARY_FIRST} />)
+                .withDriver(DatePickerDemoDriver);
 
-            await waitForDom(() => expect(bodySelect(datePickerDropdownId)).to.be.absent());
+            await waitForDom(() => expect(driver.datePicker.dropDown).to.be.absent());
 
-            simulate.focus(datePickerInput);
+            driver.datePicker.openCalender();
 
-            await waitForDom(() => expect(bodySelect(datePickerDropdownId)).to.be.present());
+            await waitForDom(() => expect(driver.datePicker.dropDown).to.be.present());
 
-            simulate.mouseDown(bodySelect('DAY_4'));
+            driver.datePicker.click(driver.datePicker.getDay(4)!);
 
             await waitForDom(() => {
-                expect(bodySelect(datePickerDropdownId)).to.be.absent();
-                expect(select(currentDate)).to.have.text('Wed Jan 04 2017');
+                expect(driver.datePicker.dropDown).to.be.absent();
+                expect(driver.date).to.have.text('Wed Jan 04 2017');
             });
         });
     });
 
     it('should only call onChange once', async () => {
         const onChange = sinon.spy();
-        const {select, waitForDom} = clientRenderer.render(<DatePicker onChange={onChange} />);
-        const datePickerInput = select(datePickerInputId);
+        const {driver, waitForDom} = clientRenderer.render(<DatePicker onChange={onChange} />)
+            .withDriver(DatePickerTestDriver);
 
-        trigger.change(datePickerInput!, '2017/02/01');
-        simulate.blur(datePickerInput);
+        driver.changeDate('2017/02/01');
 
         await waitForDom(() => expect(onChange).to.have.been.calledOnce);
     });
 
     it('should use a provided value', async () => {
-        const {select, waitForDom} = clientRenderer.render(<DatePicker value={JANUARY_FIRST} />);
+        const {driver, waitForDom} = clientRenderer.render(<DatePicker value={JANUARY_FIRST} />)
+            .withDriver(DatePickerTestDriver);
 
-        await waitForDom(() => expect(select(datePickerInputId)).to.have.value(JANUARY_FIRST.toDateString()));
+        await waitForDom(() => expect(driver.input).to.have.value(JANUARY_FIRST.toDateString()));
```

Comment:
why not have the interface `driver.currentDate`. I think it would be much more convenient